common/table.js

```diff
@@ -186,6 +203,74 @@
             $log.debug("counter", vm.flowControlObject.counter, ": after aggregated value for column (index=" + colIndex + ") update: " + (res? "successful." : "unsuccessful."));
         }
 
+        /**
+         * This will be called after the data for an aggregate column has returned.
+         * It will,
+         *  - update the templateVariables.
+         *  - update the aggregateResults
+         *  - update the column value if all of its dependent requests are back.
+         * @param  {Object} vm       vm object
+         * @param  {integer} colIndex index of aggregate column
+         * @param  {Object} values   the returned value
+         */
+        function afterReadAggregate(vm, colIndex, values) {
+            var agg = vm.reference.activeList.aggregates[colIndex];
+            var sourceDefinitions = vm.reference.table.sourceDefinitions;
+
+            values.forEach(function (val, valIndex) {
+
+                // update the templateVariables
+                if (agg.objects.length > 0 && Array.isArray(sourceDefinitions.sourceMapping[agg.column.name])) {
+                    // NOTE: not needed
```

Comment:
It looks like you expect `templateVariables` to be defined at this point. If it isn't, should we throw an error instead?

There's no harm in initializing it here, but if this case happens, it means we have an inconsistency being covered up.